tensorflow/compiler/xla/service/gpu/target_util.cc

```diff
@@ -57,11 +57,14 @@ struct TargetFunctionInfo {
                      absl::Span<const PrimitiveType> input_types_,
                      const PrimitiveType output_type_, bool use_bitcast_)
       : callee_name(callee_name_),
+        check_signature(true),
         input_types(input_types_),
         result_type(output_type_),
         use_bitcast(use_bitcast_) {}
   // Device function name.
   const string callee_name;
+  // Check signature of the device function and if needed generate casts
+  absl::optional<bool> check_signature;
```

Comment:
@sumesh13 is there any case `check_signature` may become `false`?

Reply:
Math calls dont need signature check. There is no mismatch between the call site and the function itself

Reply:
would you be able to come up with some examples where the field `check_signature` be truly necessary?

Reply:
Shfl is an example where different targets can call with either i32 or f32 and on AMDGPU  there is only f32 implementation. PTX has intrinsics for both i32/f32

Reply:
I see. But in this PR it seems relevant logic for kShflDown be tuned so `check_signature` becomes `false`. Do you plan to do it in a subsequent PR?

Reply:
or, on the other hand, change how the math calls are instantiated so `check_signature` field becomes `false`?

Reply:
right now it seems because `check_signature` would always be `true` we might be emitted more bitcodes than what's really needed. i think that's my major concern.

Reply:
Only the ones with signature provided should set the check_signature and force signature check. Even when there is check there is no extra code gen when there is mismatch such as the case for shfl_i32. I can certainly drop the signature for shfl_i32

Reply:
I think we should keep `kShflDownI32`. Instead I'd recommend change other ctors of `TargetFunctionInfo` so the initialization list becomes complete for those other cases, and we explicitly set `check_signature` be `false`, especially this one:

```
TargetFunctionInfo(const string callee_name_)
```

Reply:
also it seems this field shouldn't be `absl::optional<bool>`, but a `bool` seems more appropriate.

Reply:
and on the other hand, `use_bitcast` looks better to be modeled as `absl::optional<bool>` because it's only needed when `check_signature` is `true`?

Reply:
also, we might need to review cases in `kShflDownI32` and `kShflDownF32`. currently both call sites specify `use_bitcast` as `true`. would there be cases they should be `false` so `CreateFPtoSI` is used?